source/AST/AnyBlock.hpp

```diff
@@ -98,6 +98,47 @@ class JavadocNodesBlock
     {
         switch (ID)
         {
+        case JAVADOC_NODE_ADMONISH:
+        {
+            doc::Admonish admonish =
+                doc::Admonish::none;
+            if(auto err = decodeRecord(R, admonish, Blob))
+                return err;
+            auto node = nodes.back().get();
+            if(node->kind != doc::Kind::admonition)
+                return Error("admonish on wrong kind");
+            static_cast<doc::Admonition*>(
+                node)->style = admonish;
+            return Error::success();
+        }
+
+        case JAVADOC_PARAM_DIRECTION:
+        {
+            doc::ParamDirection direction =
+                doc::ParamDirection::none;
+            if(auto err = decodeRecord(R, direction, Blob))
+                return err;
+            auto node = nodes.back().get();
+            if(node->kind != doc::Kind::param)
+                return Error("direction on wrong kind");
+            auto param = static_cast<doc::Param*>(node);
+            param->direction = direction;
+            return Error::success();
+        }
+
+        case JAVADOC_NODE_HREF:
+        {
+            switch(auto node = nodes.back().get();
```

Comment:
I'm not a fan of single case switch statements .... we should consider removing them

Reply:
Yeah ok, remove it after we ship mrdox 2.0, when we know that we will likely not add anything more to the switch